Sage/carbon
# chore: resolve runtime errors thrown by jest tests

### Proposed behaviour

- Downgrade `@tanstack/react-virtual` to 3.10.1 to fix a runtime error in `SimpleSelect` jest tests, where the package updates state after the component unmounts.
- Fix all jest tests that are currently throwing runtime errors.
- Add `jest-fail-on-console` to fail tests on runtime errors, preventing issues from slipping into production going forward.

### Current behaviour

- Some jest tests are throwing runtime errors ([link to examples](https://github.com/Sage/carbon/actions/runs/11271858596/job/31345582300#step:5:8)) that aren't caught by CI, because jest doesn't fail on console errors.

### Checklist




#### QA


### Additional context



### Testing instructions

- Ensure there are no regressions in the virtual scrolling behaviour of the `Select` component in its Storybook stories
